### PR TITLE
update to 0.24.0

### PR DESCRIPTION
**Pull request by cbouss**
Changes:
- Update to 0.24.0
- Add tests

https://github.com/python-trio/trio/tree/v0.24.0

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,24 +23,31 @@ requirements:
   run:
     - python
     - async_generator >=1.9
```
Comment (lorepirri) [inline recipe/meta.yaml]: `async_generator >=1.9` seems needed only for tests 

https://github.com/search?q=repo%3Apython-trio%2Ftrio%20async_generator&type=code

https://github.com/python-trio/trio/blob/b9851229f095ea4aa38c470b425c1a5341fe512f/pyproject.toml#L38-L51

**Review by lorepirri — APPROVED** [review]
LGTM

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml]: Thank you! Addressed.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -22,25 +22,31 @@ requirements:
...
     - pip
+    - pytest
+    - async_generator >= 1.9
+    - astor
+    - isort
+    - pyopenssl >= 22.0.0   # for the ssl + DTLS tests
+    - trustme               # for the ssl + DTLS tests
   commands:
     - pip check
+    - python -m pytest --pyargs trio -ra --durations=10 -m "not redistributors_should_skip" --skip-optional-imports
```
Comment (danpetry) [inline recipe/meta.yaml:49]: comment about why -ra and --durations-10 are there could be useful

File: recipe/meta.yaml [hunk 3]
```diff
@@ -22,25 +22,31 @@ requirements:
...
     - outcome
-    - sniffio
+    - sniffio >=1.3.0
     - sortedcontainers
-    - exceptiongroup >=1.0.0rc9  # [py<311]
+    - exceptiongroup # [py<311]
 
 test:
+  source_files:
+    - pyproject.toml
```
Comment (danpetry) [inline recipe/meta.yaml:35]: a comment here could be useful

**Review by danpetry — APPROVED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/meta.yaml:49]: -ra gives a short summary of all except pass tests.
--durations gives the slowest 10 durations.
Upstream is using those flags, we could do without. But anycase, these are standard pytest arguments.

File: recipe/meta.yaml (hunk 3 above)
Comment (cbouss) [reply recipe/meta.yaml:35]: pytest configuration is in pyproject.toml.

File: recipe/meta.yaml (hunk 2 above)
Comment (danpetry) [reply recipe/meta.yaml:49]: I was wondering why we need e.g. the slowest 10 durations. If it's just taken from upstream, fine

File: recipe/meta.yaml (hunk 3 above)
Comment (danpetry) [reply recipe/meta.yaml:35]: ok, thanks for clarifying my understanding!

**Pull request merged**